### PR TITLE
half a gamess interface

### PR DESCRIPTION
**Pull request by loriab**
bring your own input file. known to run and parse HF (rhf/uhf/rohf) and FCI. much regex from Nuwan de Silva

### CODE REVIEW

**Comment by loriab** [comment]
The gamess interface is no further along, but I'm recommending this be merged because it's not hurting anything and, more importantly, I've standardized the `found` and `get_version` fns across the programs. Also added version stubs for cfour and nwchem. For those and gamess, have to run a miniature dummy file to get the header printed to scrape for version.

File: qcengine/programs/util/pdict.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+from decimal import Decimal, ROUND_FLOOR, ROUND_CEILING
+
+import numpy as np
+
+# This file has considerable history in the quantum chemistry package Psi4
```
Comment (dgasmith) [inline qcengine/programs/util/pdict.py]: Having a util folder and a `util.py` might be getting out of hand.

File: qcengine/programs/__init__.py [hunk 2]
```diff
@@ -63,3 +66,6 @@ def list_available_programs() -> Set[str]:
 register_program(DFTD3Executor())
 register_program(TeraChemExecutor())
 register_program(MP2DExecutor())
+register_program(GAMESSExecutor())
+register_program(NWChemExecutor())
+register_program(CfourExecutor())
```
Comment (dgasmith) [inline qcengine/programs/__init__.py]: We should not register these until they are complete.

File: qcengine/programs/cfour.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+from .executor import ProgramExecutor
+#from ...extras import provenance_stamp
+
+pp = pprint.PrettyPrinter(width=120, compact=True, indent=1)
+
+
+class CfourExecutor(ProgramExecutor):
+
+    _defaults = {
+        "name": "Cfour",
```
Comment (dgasmith) [inline qcengine/programs/cfour.py]: I always thought this was `CFOUR`?

File: qcengine/programs/gamess/__init__.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+from .runner import GAMESSExecutor
```
Comment (dgasmith) [inline qcengine/programs/gamess/__init__.py]: It doesn't seem we need a whole folder for this? No need to increase the depth if we can add everything to a single file.

Comment (loriab) [reply qcengine/programs/gamess/__init__.py]: The ones with parsing are going to get really long. What’s here for gamess is only half of a program that does  a few methods. I’d rather reconsider file layout at the end.

File: qcengine/programs/util/pdict.py (hunk 1 above)
Comment (loriab) [reply qcengine/programs/util/pdict.py]: I agree. But these are internal (non exported) utils. What about renaming the outer util “executing” in keeping with recent trends? (Would need a qcel bump)

**Pull request merged**